src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -144,13 +164,49 @@ public AdjustmentRule[] GetAdjustmentRules()
                             new DateTime(rule.DateEnd.Ticks + _baseUtcOffset.Ticks + rule.DaylightDelta.Ticks, DateTimeKind.Unspecified) :
                             rule.DateEnd;
 
-                TransitionTime startTransition = TimeZoneInfo.TransitionTime.CreateFixedDateRule(new DateTime(1, 1, 1, start.Hour, start.Minute, start.Second), start.Month, start.Day);
-                TransitionTime endTransition = TimeZoneInfo.TransitionTime.CreateFixedDateRule(new DateTime(1, 1, 1, end.Hour, end.Minute, end.Second), end.Month, end.Day);
+                if (start.Year == end.Year || !rule.NoDaylightTransitions)
```

Comment:
Can we add a comment inside this `if` for what are the scenarios that this handles? Basically, "for rules spanning less than a year, or XXX". I'm having a hard time remembering when XXX happens - when we would make a `!NoDaylightTransitions` rule on Unix. Is that only for the trailing POSIX rules past 2037?

Reply:
Sure, I'll add the comment.

> when we would make a !NoDaylightTransitions rule on Unix. Is that only for the trailing POSIX rules past 2037?

That is right. https://github.com/dotnet/runtime/blob/705daa17c134f6485830785a1cd07a27c59261a1/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs#L1125